### DISCUSSION TITLE
基于 XTuner 的 Easy-to-use 大模型可视化训练推理界面

### DISCUSSION THREAD

**Discussion by vansin** [InternLM/Tutorial#360] [project]
# XTuner 大模型可视化训练界面

**背景：** 由于 XTuner 命令行使用方式对于入门的同学依旧有着不小的上手难度，故本项目尝试使用 Gradio 搭建一个基于 XTuner 实现的 Web 界面 

**基础功能**

- GUI 可设置学习率、模型等参数生成 config
- GUI 可选择模型、可上传数据、可显示训练日志
- ....

**进阶功能**

- 显示 Loss 曲线等
- ....

![image](https://github.com/InternLM/tutorial/assets/25839884/652b9016-5fb2-4500-ac3f-17eee26f1bda)

如果实现的比较优秀与高效将会被 merge 到 XTuner 主分支

**可参考项目：**[LLaMA-Factory](https://github.com/hiyouga/LLaMA-Factory)

**Comment by SiliconSisyphus** [comment]
还加人吗？

**Reply by vansin** [reply]
还没有负责人

**Reply by vansin** [reply]
@AlwaysMissU 请私聊浦语小助手哟

**Comment by scchy** [comment]
申请负责该项目

我的进阶部署及笔记实践如下：

- HW3 langchain-部署:
  - Github: https://github.com/scchy/LLM_CookingAssistant
  - openxlab应用： https://openxlab.org.cn/apps/detail/Scchy/LLM_CookingAssistant
- HW4 xtuner-adpater-merge部署:
  - Github: https://github.com/scchy/LLM_medQA
  - openxlab应用： https://openxlab.org.cn/apps/detail/Scchy/LLM_medQA
 - 笔记&作业： https://github.com/scchy/My_Learn/tree/master/LLM_prime
